File: Backend/Models/Kontakt.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models{

    public class Kontakt{

        [Key]
        [Column("ID")]
        public int id { get; set; }

        [Column("Ime")]
        [MaxLength(30)]
        public string ime { get; set; }

        [Column("Prezime")]
        [MaxLength(30)]
        public string prezime { get; set; }

        [Column("Tip")]
        [MaxLength(30)]
        public string tip { get; set; }

        [Column("Opis")]
        [MaxLength(255)]
        public string opis { get; set; }

        public virtual List<Telefon> listaTelefona { get; set; }

    }
}

File: Frontend/kontakt.js
export class Kontakt{
    
    constructor(ime, prezime, tip, opis){
        this.id = -1;
        this.ime = ime;
        this.prezime = prezime;
        this.tip = tip;
        this.opis = opis;

        this.listaTelefona = [];
    }

    dodajTelefon(telefon){
        
        let ind = this.listaTelefona.findIndex(l => l.broj ===  telefon.broj);
        if (ind > -1){
            alert("Ovaj broj se već nalazi u listi.");
        }
        else{
            this.listaTelefona.push(telefon);
        };
    }

    izmeniKontakt(ime, prezime, tip, opis){
        this.ime = ime;
        this.prezime = prezime;
        this.tip = tip;
        this.opis = opis;
    }
}